### PR TITLE
WHFB-130: Sync Participant

### PR DESCRIPTION
**Pull request by erawat**
## Overview

This PR sync participant information to Eventify and record a response to participant custom field.  

## Before

The feature does not exist. 

## After

After the participant is saved, the hook will sync information based on the requirements and sync to the eventify by the event ID and roles that configured in the event.

### CODE REVIEW

File: Civi/Eventify/Hook/Post/EventifySync.php [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+  private function shouldSync() {
+    $rolesToSync = $this->event[$this->participantRolesToSyncField];
+
+    if (empty($rolesToSync)) {
+      return TRUE;
+    }
+
+    $participantRoles = $this->participant['participant_role_id'];
+    $hasRolesToSync = FALSE;
+    if (!is_array($participantRoles) && in_array($participantRoles, $rolesToSync)) {
```
Comment (olayiwola-compucorp) [inline Civi/Eventify/Hook/Post/EventifySync.php:182]: @erawat Are we sure this condition is valid?

File: Civi/Eventify/Hook/Post/EventifySync.php [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+      $errorCode = $response['error']['code'];
+      $errorResponse = $response['error'];
+    }
+    else {
+      $errorCode = $code;
+      $errorResponse = $response;
+    }
+
+    $this->updateParticipantSyncStatus($errorCode, $errorResponse);
+    throw new \Exception('There is an issue syncing participant ' . var_dump($response));
```
Comment (olayiwola-compucorp) [inline Civi/Eventify/Hook/Post/EventifySync.php:356]: the var_dump, is it deliberate @erawat

Comment (erawat) [reply Civi/Eventify/Hook/Post/EventifySync.php:356]: @olayiwola-compucorp  yes, sir

Comment (olayiwola-compucorp) [reply Civi/Eventify/Hook/Post/EventifySync.php:356]: great chief

**Review by olayiwola-compucorp — APPROVED** [review]

**Pull request merged**